ext/FFTWChainRulesCoreExt.jl

```diff
@@ -0,0 +1,72 @@
+module FFTWChainRulesCoreExt
+
+using FFTW
+using FFTW: r2r
+using ChainRulesCore
+
+# DCT
+
+function ChainRulesCore.frule(Δ, ::typeof(dct), x::AbstractArray, region...)
```

Comment:
IMO the call sequences in FFTW are a bit weird - it seems at some point a missing `region` is set to `1:ndims(x)` but sometimes this only happens when the corresponding plan is created.... Since we don't want to support arbitrary arguments maybe just set
```suggestion
function ChainRulesCore.frule(Δ, ::typeof(dct), x::AbstractArray, region=1:ndims(x))
```
or use a default of `nothing` and branch in the function body?

Reply:
done!